estate/models/estate_property_offer.py

```diff
@@ -0,0 +1,65 @@
+from datetime import datetime,timedelta
+from odoo import api,fields, models
+from odoo.exceptions import UserError,ValidationError
+
+class EstatePropertyOffer(models.Model):
+    _name = "estate.property.offer"
+    _description = "Estate Property Offer Model"
+    _order = "price desc"
+    _sql_constraints = [('price_pos','check(price>0)','price should be positive!')]
+
+    price = fields.Float()
+    state = fields.Selection(
+        selection = [('accepted','Accepted'),('refused','refused')]
+    )
+    validity = fields.Integer(default=7)
+    deadline = fields.Date(compute="_compute_date_deadline",inverse="_inverse_deadline",store=True)
+    partner_id = fields.Many2one('res.partner',required=True)
+    property_id = fields.Many2one('estate.property',required=True)
+    property_type_id  = fields.Many2one(related='property_id.property_type_id', store=True)    
+    
+
+    @api.depends('validity')
+    def _compute_date_deadline(self):
+        for record in self:
+            if record.create_date:
+                record.deadline = record.create_date.date()+ timedelta(days=record.validity)
+            else:
+                record.deadline = fields.date.today()+ timedelta(days=record.validity)
```

Comment:
```suggestion
            record.deadline = record.create_date.date() or fields.date.today() + timedelta(days=record.validity)
```
by using or it will take the first if found else second one.
but it's somewhat optimized

Reply:
then it should be like this 
` record.deadline = (record.create_date and record.create_date.date() or fields.date.today()) + timedelta(days=record.validity) `
other wise there will be error because we are not check boolean value directly accessing date object which will show bool has no object